src/ts/pages/DataCatalog/__tests__/DataCatalog.test.tsx

```diff
@@ -26,27 +26,75 @@ window.scrollTo = vi.fn()
 
 const mockedUseMediaQuery = useMediaQuery as ReturnType<typeof vi.fn>
 
-function joinUrl(url :string, params: string) {
+function joinUrl(url: string, params: string) {
   // Split the URL into the base part and the query string
   const [baseUrl, queryString] = url.split('?')
 
-  const urlParams: Record<string, string> = {}
+  const urlParams: Record<string, string | string[]> = {}
+
+  // Parse existing query string
   if (queryString) {
     queryString.split('&').forEach((pair) => {
       const [key, value] = pair.split('=')
-      urlParams[key] = value
+      const decodedKey = decodeURIComponent(key)
+      const decodedValue = decodeURIComponent(value || '')
+
+      if (decodedKey.endsWith('[]')) {
+        // Handle array parameters
+        const arrayKey = decodedKey.slice(0, -2)
+        if (!urlParams[arrayKey]) {
+          urlParams[arrayKey] = []
+        }
+
+        if (Array.isArray(urlParams[arrayKey])) {
+          (urlParams[arrayKey] as string[]).push(decodedValue)
+        }
+      } else {
+        urlParams[decodedKey] = decodedValue
+      }
     })
   }
```

Comment:
function joinUrl(url: string, params: string) {
  const [baseUrl, queryString] = url.split('?')

  // Parse existing parameters
  const existingParams = new URLSearchParams(queryString || '')

  // Parse new parameters
  const newParams = new URLSearchParams(params)

  // Merge parameters (new params override existing ones)
  newParams.forEach((value, key) => {
    existingParams.set(key, value)
  })

  return `${baseUrl}?${existingParams.toString()}`
}

Reply:
This one is shorter and just uses node methods